problem2.java

```diff
@@ -0,0 +1,53 @@
+class Solution1 {
```

Comment:
Please mention the following:
1. Give a 3 line explanation of the approach
2. Time and Space complexities for your approach with 1 line justification for the same
3. Mention if your solution ran on leetcode